api/src/main/scala/play/twirl/api/Content.scala

```diff
@@ -57,4 +57,11 @@ abstract class BufferedContent[A <: BufferedContent[A]](protected val elements:
 
   def body = builtBody
 
+  override def equals(obj: Any): Boolean = obj match {
+    case other: BufferedContent[_] => body == other.body
```

Comment:
Right, added a check for the same class in the `equals` method and also in the `hashCode` method.

Tests also added for equality checking (let me know if you want tests for `hashCode` as well)